src/states/elder/mod.rs

```diff
@@ -1551,6 +1531,15 @@ impl Approved for Elder {
 
         let self_sec_update = elders_info.prefix().matches(self.name());
 
+        // Poll the relocate queue before the parsec reset, so it is not blocked waiting for the
+        // genesis event. Cast the actual votes only after the parsec reset however, so they already
+        // go to the new instance.
+        let relocate_details = if self_sec_update {
```

Comment:
That seem a bit weird, But I guess it can be fixed when poll relocation happen in a single place.

Reply:
It's a quick and simple fix. We can always refactor later.